safehttp/plugins/angular_xsrf/angular_xsrf.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package xsrf provides a safehttp.Interceptor that ensures Cross-Site Request
+// Forgery protection by verifying the incoming requests, rejecting those
+// requests that are suspected to be part of an attack.
+package angularxsrf
```

Comment:
Since we expect to have a couple of these packages I would call them xsrf/angular and xsrf/html in the directory tree, and I would name the package xsrf* not *xsrf.

Moreover since we are creating a new directory you could create a common package with exported helpers (e.g. a function that checks if a method is state-preserving)

The final directory tree that I envision is:
```
plugins/
  xsrf/
    angular/
      package_xsrfangular.go
    vanilla/
      package_xsrfhtml.go
    package_xsrf.go
```
and the common code between the two would go in the xsrf package. The fact that we didn't find a proper abstraction doesn't mean we have to duplicate code :)